docs/reference/core/zio/zio.md

```diff
@@ -337,24 +338,48 @@ val printLine2: IO[IOException, String] =
 ##### Blocking Synchronous Side-Effects
 
 | Function                    | Input Type                          | Output Type                     |
-| --------------------------- | ----------------------------------- | ------------------------------- |
+|-----------------------------|-------------------------------------|---------------------------------|
 | `blocking`                  | `ZIO[R, E, A]`                      | `ZIO[R, E, A]`                  |
 | `attemptBlocking`           | `A`                                 | `RIO[Blocking, A]`              |
 | `attemptBlockingCancelable` | `effect: => A`, `cancel: UIO[Unit]` | `RIO[Blocking, A]`              |
 | `attemptBlockingInterrupt`  | `A`                                 | `RIO[Blocking, A]`              |
 | `attemptBlockingIO`         | `A`                                 | `ZIO[Blocking, IOException, A]` |
 
-Some side-effects use blocking IO or otherwise put a thread into a waiting state. If not carefully managed, these side-effects can deplete threads from our application's main thread pool, resulting in work starvation.
+By default, ZIO is asynchronous and all effects will be executed on a default primary thread pool which is optimized for asynchronous operations. As ZIO uses a fiber-based concurrency model, if we run **Blocking I/O** or **CPU Work** workloads on a primary thread pool, they are going to monopolize all threads of **primary thread pool**.
+
+ZIO has a separate **blocking thread pool** specially designed for **Blocking I/O** and, also **CPU Work** workloads. We should run blocking workloads on this thread pool by using `ZIO.blocking` or `ZIO.attemptBlocking*` constructors to prevent interfering with the primary thread pool.
+
+:::note
+ZIO has an auto-blocking mechanism that detects blocking operations and runs them on a separate blocking thread pool. However, it is recommended to import blocking side effects.
```

Comment:
"side effect" terminology is deprecated and should be removed, here and elsewhere.

Reply:
@jdegoes Can you elaborate more on this?

Reply:
@khajavi The concept of "side effects" comes from functional programming and does not make sense to developers coming from other backgrounds. We have taken significant steps to make this concept not relevant. For example `ZIO.succeed` accepts a by name parameter so you use `ZIO.succeed` whether the body is side effecting or not. We are trying to "dissolve" the concept of side effects.

So I would rewrite this along the lines of:

```
ZIO has an auto-blocking mechanism that detects blocking operations and runs them on a separate blocking thread pool. However, it if you know that some code is blocking you can signal this to the ZIO runtime by using the `ZIO.blocking`
```